ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/ForeignKeyTableTableJoinBuilder.java

```diff
@@ -40,8 +45,32 @@ private ForeignKeyTableTableJoinBuilder() {
     final LogicalSchema leftSchema = left.getSchema();
     final LogicalSchema rightSchema = right.getSchema();
 
+    final ProcessingLogger logger = buildContext.getProcessingLogger(
+        join.getProperties().getQueryContext()
+    );
+
+    final Optional<ExpressionEvaluator> expressionEvaluator;
+
+    final Optional<Expression> leftJoinExpression = join.getLeftJoinExpression();
+    if (leftJoinExpression.isPresent()) {
+      final CodeGenRunner codeGenRunner = new CodeGenRunner(
+          leftSchema,
+          buildContext.getKsqlConfig(),
+          buildContext.getFunctionRegistry()
+      );
+
+      expressionEvaluator = Optional.of(
+          codeGenRunner.buildCodeGenFromParseTree(
+              leftJoinExpression.get(),
+              "Left Join Expression"
+          )
+      );
+    } else {
+      expressionEvaluator = Optional.empty();
```

Comment:
Instead of having two separate cases here, can we combine them by always passing an expression evaluator in order to simplify the code? The expression evaluator for the single column case would be simple, just extracting the appropriate column.

Reply:
Good idea!